examples/bond/scripts/run/bond-functions.js

```diff
@@ -0,0 +1,328 @@
+const {
+  executeTransaction, convert, readGlobalStateSSC, balanceOf
+} = require('@algo-builder/algob');
+const { types } = require('@algo-builder/web');
+const { exec } = require('child_process');
+const { exit } = require('process');
+
+let newAsaInfo;
+let appInfo;
+let asaInfo;
+let buybackLsig;
+const assetID = 'asset-index';
+
+// fund account using master account
+async function fundAccount (deployer, accountAddress) {
+  const masterAccount = deployer.accountsByName.get('master-account');
+  const algoTxnParams = {
+    type: types.TransactionType.TransferAlgo,
+    sign: types.SignType.SecretKey,
+    fromAccount: masterAccount,
+    toAccountAddr: accountAddress,
+    amountMicroAlgos: 200e6,
+    payFlags: {}
+  };
+  await executeTransaction(deployer, algoTxnParams);
+}
+
+/**
+ * Creates DEX_i lsig, burn B_i tokens, issue B_i+1 tokens
+ * @param {Account} masterAccount
+ * @param {Account} creatorAccount
+ * @param {Account} storeManagerAccount
+ */
+async function createDex (deployer, masterAccount, creatorAccount, storeManagerAccount) {
+  asaInfo = deployer.getASAInfo('bond-token');
+  appInfo = deployer.getApp('bond-dapp-stateful.py', 'bond-dapp-clear.py');
+  let scInitParam = {
+    TMPL_APPLICATION_ID: appInfo.appID,
+    TMPL_OWNER: creatorAccount.addr,
+    TMPL_STORE_MANAGER: storeManagerAccount.addr
+  };
+  const issuerLsig = await deployer.loadLogic('issuer-lsig.py', scInitParam);
+
+  const deployTx = {
+    type: types.TransactionType.DeployASA,
+    sign: types.SignType.SecretKey,
+    fromAccount: creatorAccount,
+    asaName: 'new-bond-token',
+    payFlags: {}
+  };
+  // Create B_[i+1]
+  newAsaInfo = await executeTransaction(deployer, deployTx);
+  console.log(newAsaInfo);
+  const newIndex = newAsaInfo[assetID];
+
+  // Only store manager can allow opt-in to ASA for lsig
+  const optInTx = [
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: storeManagerAccount,
+      toAccountAddr: issuerLsig.address(),
+      amountMicroAlgos: 0,
+      payFlags: {}
+    },
+    {
+      type: types.TransactionType.OptInASA,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: issuerLsig.address(),
+      lsig: issuerLsig,
+      assetID: newIndex,
+      payFlags: {}
+    }
```

Comment:
this is same transaction as in deploy. Let's export it to a common function and reuse.

Reply:
ok